gusto/time_discretisation/time_discretisation.py

```diff
@@ -219,31 +223,42 @@ def setup(self, equation, apply_bcs=True, *active_labels):
 
             else:
                 if self.wrapper_name == "supg":
-                    self.wrapper.setup()
+                    if self.wrapper_field_names is not None:
+                        for field_name in self.wrapper_field_names:
+                            self.wrapper.setup(field_name)
+                            new_test = self.wrapper.test
+                            self.residual = self.residual.label_map(
+                                lambda t: t.get(prognostic) == field_name and t.has_label(transport),
```

Comment:
I generally don't like specific terms appearing in this file because I think the `time_discretisation` should be agnostic to the type of term being solved...

... if we need to specifically only replace the test function in the transport terms, could the `SUPGOptions`  object contain a list of labels for the terms that need replacing? Then the user can specify which terms are going to be adjusted

Reply:
Okay agreed, I have added a list of term labels to cycle through. I need to add a test for this so hang fire for that